### ISSUE TITLE
`ignore_extra_fields()` does not ignore objects

### ISSUE THREAD

**Issue by ThijsJung** [williamwilling/okay#1]
# Context
I am using Okay to validate deeply nested Python objects of furniture data. Part of this data is shared by all types of furniture (dimensions, prices, colors, etc) but some attributes are specific to the type of furniture; e.g. a chair can have an armrest height while a lamp will not, and a lamp can have a cable length while a sofa will most probably not. I wanted to use separate schemas for different product types and in those schemas I wanted to ignore the extra fields. While trying to implement that I noticed extra fields that where not ignored.

# Observed issue
I adapted the `ignore_extra_fields()` example [from the user guide](https://github.com/williamwilling/okay/blob/master/docs/user-guide.md#unspecified-fields) to show what I mean. I added a `published_at` object to the book object and ran the validation.

```
from okay import validate
from okay.schema import *

def schema():
    ignore_extra_fields()

book = {
    'title': True,
    'author': [ 'Vikram' , 'Seth' ],
    'published_at': {
        'year': 1999,
        'month': 4,
        'day': 8
    }
}

validation_messages = validate(schema, book)
for message in validation_messages:
    print(f'{message.field}:\t{message.type}')
```

Output:
```
published_at.year:      extra_field
published_at.month:     extra_field
published_at.day:       extra_field
```

I didn't expect this to happen. Is there something I misunderstood? Thanks in advance for helping out!

**Comment by williamwilling** [comment]
This bug got fixed unwittingly in 77e5a5e3051fc44732691f84ac112c946cf39851, but it has been lingering on the `develop` branch for a long time, along with a couple of useful features. I've now released version 2.0.0 of Okay. As the version number suggests, there are some breaking changes, so check out the [changelog](https://github.com/williamwilling/okay/blob/v2/docs/changelog.md). If this is a problem for you and you need me to fix the bug in v1, let me know.

**Comment by ThijsJung** [comment]
I didn't experience any hiccups while upgrading to version 2.0.0, so no need for me to fix the bug in v1. Also, kudos on the new features, very cool! Thanks for picking this up on such short notice. 😊
